app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java

```diff
@@ -96,71 +103,79 @@ public MongoPlugin(PluginWrapper wrapper) {
+                                /**
+                                 * The json contains key "cursor" when find command was issued and there are 1 or more
+                                 * results. In case there are no results for find, this key is not present in the result json.
+                                 */
+                                if (outputJson.has("cursor")) {
+                                    JSONArray outputResult = (JSONArray) cleanUp(
+                                            outputJson.getJSONObject("cursor").getJSONArray("firstBatch"));
+                                    result.setBody(objectMapper.readTree(outputResult.toString()));
+                                }
+
+                                /**
+                                 * The json contains key "n" when insert/update command is issued. "n" for update
+                                 * signifies the no of documents selected for update. "n" in case of insert signifies the
+                                 * number of documents inserted.
+                                 */
+                                if (outputJson.has("n")) {
+                                    JSONObject body = new JSONObject().put("n", outputJson.getBigInteger("n"));
+                                    result.setBody(body);
+                                    headerArray.put(body);
+                                }
+
+                                /**
+                                 * The json key contains key "nModified" in case of update command. This signifies the no of
+                                 * documents updated.
+                                 */
+                                if (outputJson.has(N_MODIFIED)) {
+                                    JSONObject body = new JSONObject().put(N_MODIFIED, outputJson.getBigInteger(N_MODIFIED));
+                                    result.setBody(body);
+                                    headerArray.put(body);
+                                }
+
+                                /** TODO
+                                 * Go through all the possible fields that are returned in the output JSON and add all the fields
+                                 * that are important to the headerArray.
+                                 */
+                            }
 
-                JSONObject statusJson = new JSONObject().put("ok", status);
-                headerArray.put(statusJson);
-                result.setHeaders(objectMapper.readTree(headerArray.toString()));
-            } catch (Exception e) {
-                return Mono.error(new AppsmithPluginException(AppsmithPluginError.PLUGIN_ERROR, e));
-            }
+                            JSONObject statusJson = new JSONObject().put("ok", status);
+                            headerArray.put(statusJson);
+                            result.setHeaders(objectMapper.readTree(headerArray.toString()));
+                        } catch (IOException e) {
```

Comment:
Let's move back to generic exception to ensure that we throw a plugin error for all scenarios. If any other exception is thrown, we would end up throwing an internal server error instead of plugin error. 
```suggestion
                        } catch (Exception e) {
```

Reply:
Done.